modules/pam_env/pam_env.c

```diff
@@ -581,9 +581,9 @@ _expand_arg(pam_handle_t *pamh, char **value)
 		    * a constant so that the compiler will shut up when I
 		    * call pam_getenv and _pam_get_item_byname -- sigh
 		    */
+  const char *tmpval;
```

Comment:
While this change is correct, I was a bit confused by the commit message.
If I understand correctly, getting rid of the copy is OK because during the short window when the copy is used the original string is _not_ modified, and the fact that the original string is modified afterwards doesn't look relevant for this change.

Reply:
I have simplified the commit message. I have tried to point out that the const keyword is kind of violated on purpose (which the comment in the code remotely hints at), but since it is really not relevant to the change, it is dropped.